### PR TITLE
Add AppSessionId

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
* Event API - Agreement events
* Add AppSessionId

resolves: #726

Contains changes from:  #754
which resolves: #729

### CODE REVIEW

File: core/market/migrations/2020-10-04-190200_agreement_events/up.sql [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    session_id VARCHAR(100),
+
+    creation_ts DATETIME NOT NULL,
+    valid_to DATETIME NOT NULL,
+    state VARCHAR(20) NOT NULL,
+    approved_date DATETIME,
+
+    proposed_signature TEXT,
+    approved_signature TEXT,
+    committed_signature TEXT
```
Comment (tworec) [inline core/market/migrations/2020-10-04-190200_agreement_events/up.sql:49]:
```suggestion
    committed_signature TEXT,

    CHECK (state in ('Proposal','Pending','Cancelled','Rejected','Approved','Expired','Terminated'))
```

File: core/market/migrations/2020-10-04-190200_agreement_events/up.sql [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    provider_id VARCHAR(20) NOT NULL,
+    requestor_id VARCHAR(20) NOT NULL,
+
+    session_id VARCHAR(100),
+
+    creation_ts DATETIME NOT NULL,
+    valid_to DATETIME NOT NULL,
+    state VARCHAR(20) NOT NULL,
+    approved_date DATETIME,
```
Comment (tworec) [inline core/market/migrations/2020-10-04-190200_agreement_events/up.sql:45]: btw.  maybe we can make the name consistent
```suggestion
    approved_ts DATETIME,
```

File: core/market/src/db/dao.rs [hunk 3]
```diff
@@ -10,11 +10,13 @@ pub mod sql_functions {
         fn coalesce_id(column: sql_types::Nullable<sql_types::Text>, default: sql_types::Text) -> sql_types::Text
     );
 }
+mod agreement_events;
```
Comment (tworec) [inline core/market/src/db/dao.rs:13]: nitpick: this belongs rather on line 2. PLs sort modules

File: core/market/src/db/dao/agreement.rs [hunk 4]
```diff
@@ -58,12 +68,12 @@ impl<'c> AgreementDao<'c> {
     ) -> DbResult<Option<Agreement>> {
         let id = id.clone();
         do_with_transaction(self.pool, move |conn| {
-            let mut query = dsl::market_agreement.filter(dsl::id.eq(&id)).into_boxed();
+            let mut query = market_agreement.filter(agreement::id.eq(&id)).into_boxed();
```
Comment (tworec) [inline core/market/src/db/dao/agreement.rs:71]: 👍

File: core/market/src/db/dao/agreement.rs [hunk 5]
```diff
@@ -113,10 +117,52 @@ impl<'c> AgreementDao<'c> {
...
+                    from: agreement.state,
+                    to: AgreementState::Pending,
+                })?
+            }
+
+            update_state(conn, &id, &AgreementState::Pending)
+                .map_err(|e| StateError::DbError(e.into()))?;
+
+            if let Some(session) = session {
+                diesel::update(market_agreement.filter(agreement::id.eq(&id)))
```
Comment (tworec) [inline core/market/src/db/dao/agreement.rs:144]: In `update_state` you are using shorter form. Is there a difference?
```suggestion
                diesel::update(market_agreement.find(id))
```

File: core/market/src/db/dao/agreement.rs [hunk 6]
```diff
@@ -126,31 +172,59 @@ impl<'c> AgreementDao<'c> {
...
+
+            // It's important, that if None AppSessionId comes, we shouldn't update Agreement
+            // appSessionId field to None. This function can be called in different context, for example
+            // on Requestor, when appSessionId is already set.
+            if let Some(session) = session {
+                diesel::update(market_agreement.filter(agreement::id.eq(&id)))
+                    .set(agreement::session_id.eq(session))
+                    .execute(conn)
+                    .map_err(|e| StateError::SessionId(e.into()))?;
+            }
```
Comment (tworec) [inline core/market/src/db/dao/agreement.rs:186]: deserves wrapping into separate fn

File: core/market/src/db/dao/agreement.rs [hunk 7]
```diff
@@ -126,31 +172,59 @@ impl<'c> AgreementDao<'c> {
...
 
-            diesel::update(dsl::market_agreement.filter(dsl::id.eq(&id)))
-                .set(dsl::state.eq(AgreementState::Approved))
-                .execute(conn)?;
+            update_state(conn, &id, &AgreementState::Approved)
+                .map_err(|e| StateError::DbError(e.into()))?;
+
+            // It's important, that if None AppSessionId comes, we shouldn't update Agreement
+            // appSessionId field to None. This function can be called in different context, for example
+            // on Requestor, when appSessionId is already set.
```
Comment (tworec) [inline core/market/src/db/dao/agreement.rs:180]: 👍

File: core/market/src/db/dao/agreement.rs [hunk 8]
```diff
@@ -126,31 +172,59 @@ impl<'c> AgreementDao<'c> {
                 })?
             }
 
-            diesel::update(dsl::market_agreement.filter(dsl::id.eq(&id)))
-                .set(dsl::state.eq(AgreementState::Approved))
-                .execute(conn)?;
+            update_state(conn, &id, &AgreementState::Approved)
+                .map_err(|e| StateError::DbError(e.into()))?;
```
Comment (tworec) [inline core/market/src/db/dao/agreement.rs:176]: important: we also want to set `approved_date` here! (or `approved_ts` if you decide to refactor it)

File: core/market/src/db/dao/agreement.rs [hunk 9]
```diff
@@ -126,31 +172,59 @@ impl<'c> AgreementDao<'c> {
...
+            let num_events = diesel::delete(related_events).execute(conn)?;
+            Result::<(usize, usize), DbError>::Ok((num_agreements, num_events))
         })
         .await?;
-        if num_deleted > 0 {
-            log::info!("Clean market agreements: {} cleaned", num_deleted);
+
+        if num_agreements > 0 {
+            log::info!("Clean market agreements: {} cleaned", num_agreements);
+            log::info!("Clean market agreement events: {} cleaned", num_events);
```
Comment (tworec) [inline core/market/src/db/dao/agreement.rs:225]:
```suggestion
            log::info!("Cleaned {} market agreements", num_agreements);
            log::info!("Cleaned {} market agreement events", num_events);
```

File: core/market/src/db/dao/agreement_events.rs [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        session_id: &AppSessionId,
+        max_events: i32,
+        after_timestamp: NaiveDateTime,
+    ) -> DbResult<Vec<AgreementEvent>> {
+        let session_id = session_id.clone();
+        let node_id = node_id.clone();
+        readonly_transaction(self.pool, move |conn| {
+            // We will get only one Agreement, by using this filter.
+            // There will be no way to get Requestor'a Agreement, when being Provider, and vice versa,
+            // because AgreementId for Provider and Requestor in Agreement Event table is different.
```
Comment (tworec) [inline core/market/src/db/dao/agreement_events.rs:39]: 👍

File: core/market/src/db/dao/agreement_events.rs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            let filter_out_not_my_agreements = agreement::provider_id
```
Comment (tworec) [inline core/market/src/db/dao/agreement_events.rs:40]:
```suggestion
            let filter_my_agreements = agreement::provider_id
```

File: core/market/src/db/dao/agreement_events.rs [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+                .eq(node_id)
+                .or(agreement::requestor_id.eq(node_id));
+
+            let mut select_corresponding_agreement = market_agreement
+                .select(agreement::id)
+                .filter(filter_out_not_my_agreements)
```
Comment (tworec) [inline core/market/src/db/dao/agreement_events.rs:46]:
```suggestion
                .filter(filter_my_agreements)
```

File: core/market/src/db/dao/agreement_events.rs [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+                .into_boxed();
+
+            // If AppSessionId is None, we just select all events, independent from AppSessionId
+            // set in Agreement. If AppSessionId is not None we get select only events for this id.
```
Comment (tworec) [inline core/market/src/db/dao/agreement_events.rs:50]:
```suggestion
            // Optionally filter by `AppSessionId`.
```

File: core/market/src/db/model/agreement.rs [hunk 14]
```diff
@@ -14,42 +8,45 @@ use ya_client::model::market::agreement::{
 use ya_client::model::market::demand::Demand as ClientDemand;
 use ya_client::model::market::offer::Offer as ClientOffer;
 use ya_client::model::{ErrorMessage, NodeId};
+use ya_diesel_utils::DatabaseTextField;
 
 use crate::db::model::{OwnerType, Proposal, ProposalId, SubscriptionId};
 use crate::db::schema::market_agreement;
 
 pub type AgreementId = ProposalId;
+pub type AppSessionId = Option<String>;
```
Comment (tworec) [inline core/market/src/db/model/agreement.rs:17]: as this is user input, we need to check if this is at most 100 chars. Is there a unit test for such a case?

File: core/market/src/db/model/agreement.rs [hunk 15]
```diff
@@ -14,42 +8,45 @@ use ya_client::model::market::agreement::{
...
 
 /// TODO: Could we avoid having separate enum type for database
 ///  and separate for client?
 #[derive(
-    FromPrimitive,
+    strum_macros::EnumString,
+    DatabaseTextField,
```
Comment (tworec) [inline core/market/src/db/model/agreement.rs:23]: yeah! :)

File: core/market/src/db/model/agreement_events.rs [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+        let agreement_id = self.agreement_id.into_client();
+        let event_date = DateTime::<Utc>::from_utc(self.timestamp, Utc);
+        let reason = self
+            .reason
+            .map(|reason| serde_json::from_str::<JsonReason>(&reason))
+            .map(|result| result.map_err(|e| {
+                log::warn!(
+                    "Agreement Event with not parsable Reason in database. Error: {}. Shouldn't happen \
+                     because market is responsible for rejecting invalid Reasons.", e
+                )
```
Comment (tworec) [inline core/market/src/db/model/agreement_events.rs:62]: minor: I'd also return this warning mesage as a proper JsonReason for the clients to be aware and can report if it happens one day

File: Cargo.toml [hunk 17]
```diff
@@ -185,4 +186,5 @@ ya-utils-actix = { path = "utils/actix_utils"}
 ya-utils-futures = { path = "utils/futures"}
 ya-utils-path = { path = "utils/path"}
 ya-utils-process = { path = "utils/process"}
+ya-diesel-utils = { path = "utils/diesel-utils"}
```
Comment (tworec) [inline Cargo.toml:189]: you should update also ya-client to have `/agreementEvents` instead `/agreements/events`

**Review by tworec — CHANGES_REQUESTED** [review]
part 1

File: core/market/tests/test_app_session_id.rs [hunk 18]
```diff
@@ -0,0 +1,549 @@
...
+async fn test_session_negotiation_on_the_same_node() -> Result<()> {
+    let network = MarketsNetwork::new(None)
+        .await
+        .add_market_instance("Node")
+        .await?;
+
+    let req_market = network.get_market("Node");
+    let req_id = network.get_default_id("Node");
+    let prov_id = network.get_default_id("Node");
+    let prov_market = network.get_market("Node");
```
Comment (tworec) [inline core/market/tests/test_app_session_id.rs:241]:
```suggestion
    let prov_id = req_id.clone();
    let prov_market = req_market.clone();
```
will do?

File: core/market/tests/test_app_session_id.rs [hunk 19]
```diff
@@ -0,0 +1,549 @@
...
+
+    let negotiation = negotiate_agreement(
+        &network,
+        "Node",
+        "Node",
+        "negotiation",
+        "r-session",
+        "p-session",
```
Comment (tworec) [inline core/market/tests/test_app_session_id.rs:249]: we should also test edge case when sessions for both sides are equal

File: core/market/tests/test_app_session_id.rs [hunk 20]
```diff
@@ -0,0 +1,549 @@
...
+    assert_eq!(p_events.len(), 1);
+    assert_eq!(r_events.len(), 1);
+    Ok(())
+}
+
+/// We expect to get only events after specified timestamp.
+#[cfg_attr(not(feature = "test-suite"), ignore)]
+#[actix_rt::test]
+#[serial_test::serial]
+async fn test_session_timestamp_filtering() -> Result<()> {
```
Comment (tworec) [inline core/market/tests/test_app_session_id.rs:287]: wow, thats huge test. I dare to propose even more: test mixed timestamp & session filtering. but will understand if you are fed up with this ;)

**Review by tworec — CHANGES_REQUESTED** [review]
uhh, it was big. But here it is

File: core/market/src/db/dao/agreement.rs (hunk 5 above)
Comment (nieznanysprawiciel) [reply core/market/src/db/dao/agreement.rs:144]: I'm not sure, if there's a difference. I didn't wrote this part of `update_state`.

Comment (nieznanysprawiciel) [reply core/market/src/db/dao/agreement.rs:144]: Hmm I checked. `find` will return single record and `filter` can return multiple records. So this will be correct here, I will change.

File: core/market/src/db/dao/agreement.rs (hunk 8 above)
Comment (nieznanysprawiciel) [reply core/market/src/db/dao/agreement.rs:176]: This is bigger change, than you might think. We need `approved_date` to match on both Provider and Requestor side, because we want to sign them together with whole Agreement. So changes in negotiation protocol are needed.
As long as we have `None` here and we don't lie to users, this can wait.
Issue for this:  #814

File: core/market/src/db/model/agreement.rs (hunk 14 above)
Comment (nieznanysprawiciel) [reply core/market/src/db/model/agreement.rs:17]: I didn't know, we have such a limit. Is it documented?

File: Cargo.toml (hunk 17 above)
Comment (nieznanysprawiciel) [reply Cargo.toml:189]: Will do in #730

File: core/market/src/utils/display.rs [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        self.0.fmt(f)
+    }
+}
+
+impl<'a> std::fmt::Display for DisplayEnabler<'a, Identity> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        write!(f, "'{}' [{}]", &self.0.name, &self.0.identity)
+    }
+}
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/utils/display.rs:39]: Do you think, this implementation of Display is suitable for all contexts? I decided here to omit `role` from Identity and someone could need it. If I will implement this in Identity crate, someone will change it and we will see thinks, that we don't want to see.

File: utils/diesel-utils/src/lib.rs [hunk 22]
```diff
@@ -0,0 +1,40 @@
+extern crate proc_macro;
+use proc_macro2::Span;
+use quote::quote;
+use syn::Error;
+
+#[proc_macro_derive(DatabaseTextField)]
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply utils/diesel-utils/src/lib.rs:6]: You took away the most important information from this name

File: core/market/src/negotiation/common.rs [hunk 23]
```diff
@@ -141,6 +165,66 @@ impl CommonBroker {
...
+                        Err(AgreementEventsError::Internal(error.to_string()))
+                    }
+                    NotifierError::Unsubscribed(_) => Err(AgreementEventsError::Internal(format!(
+                        "Code logic error. Shouldn't get Unsubscribe in Agreement events notifier."
+                    ))),
+                };
+            }
+            // Ok result means, that event with required sessionId id was added.
+            // We can go to next loop to get this event from db. But still we aren't sure
+            // that list won't be empty, because other query_agreement_events calls can wait for the same event.
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/common.rs:224]: Copy-paste. Thanks

File: core/market/src/negotiation/common.rs [hunk 24]
```diff
@@ -322,10 +421,21 @@ pub fn validate_match(
     }
 }
 
-pub struct DisplayIdentity<'a>(pub &'a Identity);
-
-impl<'a> fmt::Display for DisplayIdentity<'a> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "'{}' [{}]", &self.0.name, &self.0.identity)
+pub fn expect_state(
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/common.rs:424]: Your name isn't true. This function doesn't check transition. It checks current state, if it matches expected state.

File: core/market/src/negotiation/requestor.rs [hunk 25]
```diff
@@ -482,7 +487,7 @@ pub async fn proposal_receiver_thread(
                 .save_initial_proposal(proposal)
                 .await?;
 
-            log::debug!(
+            log::info!(
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/requestor.rs:490]: It is important to connect Offer/Demand subscriptions with generated Proposal id. We can move it back to debug, if it will spam to much (Or I can even revert it now if you insist). But I completely disagree with idea, that dwarf messages like in line 481 (prev 476) deserve to be on info level, because they are short and meaningful logs should be on debug.

**Review by Wiezzel — APPROVED** [review]

File: core/market/tests/test_agreement_events.rs [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+
+        // Expect single event
+        assert_eq!(events.len(), 1);
+
+        match &events[0] {
+            AgreementEvent::AgreementApprovedEvent { agreement_id, .. } => {
+                assert_eq!(agreement_id, &agr_id.into_client())
+            }
+            _ => panic!("Expected AgreementEvent::AgreementApprovedEvent"),
+        };
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/tests/test_agreement_events.rs:75]: I always follow the rule, that everything that is direct subject of test, should be directly visible in test without using any helpers for this (even if that means ugly code). Helpers can change, test shouldn't, unless you consciously want to change it.
Note that in other cases, where I'm using `expect_approve`, it checks semantically, what the name says - it expect approval. In this test we don't expect approval, but we want to match AgreementEvent enum.

File: core/market/tests/test_agreement_events.rs [hunk 27]
```diff
@@ -0,0 +1,175 @@
...
+    // Expect single event
+    assert_eq!(events.len(), 1);
+
+    let id = agreement_id.into_client();
+    match &events[0] {
+        AgreementEvent::AgreementApprovedEvent { agreement_id, .. } => {
+            assert_eq!(agreement_id, &id)
+        }
+        _ => panic!("Expected AgreementEvent::AgreementApprovedEvent"),
+    };
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/tests/test_agreement_events.rs:92]: Check comment above

File: core/market/tests/test_app_session_id.rs (hunk 20 above)
Comment (nieznanysprawiciel) [reply core/market/tests/test_app_session_id.rs:287]: Yes I am :P
Added to task: #781

File: core/market/src/config.rs [hunk 28]
```diff
@@ -20,6 +21,10 @@ pub struct SubscriptionConfig {
     pub default_ttl: chrono::Duration,
 }
 
+pub struct EventsConfig {
+    pub max_agreement_events: i32,
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/config.rs:25]: Ok, but I'm not adding `max_events_min`. It doesn't make sense, because it should be always 1.

File: core/market/src/config.rs [hunk 29]
```diff
@@ -38,3 +43,11 @@ impl Default for SubscriptionConfig {
         }
     }
 }
+
+impl Default for EventsConfig {
+    fn default() -> Self {
+        EventsConfig {
+            max_agreement_events: 20,
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/config.rs:50]: Same as above

File: core/market/src/negotiation/common.rs [hunk 30]
```diff
@@ -108,11 +132,11 @@ impl CommonBroker {
             return Ok(vec![]);
         }
 
-        let mut notifier = self.notifier.listen(subscription_id);
+        let mut notifier = self.negotiation_notifier.listen(subscription_id);
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/common.rs:135]: Thanks. I forgot

File: core/market/src/negotiation/common.rs [hunk 31]
```diff
@@ -141,6 +165,66 @@ impl CommonBroker {
...
+        timeout: f32,
+        max_events: Option<i32>,
+        after_timestamp: DateTime<Utc>,
+        id: &Identity,
+    ) -> Result<Vec<AgreementEvent>, AgreementEventsError> {
+        let mut timeout = Duration::from_secs_f32(timeout.max(0.0));
+        let stop_time = Instant::now() + timeout;
+        let max_events = max_events.unwrap_or(self.config.events.max_agreement_events);
+
+        if max_events <= 0 {
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/common.rs:180]: I don't add `max_events_min`, because I don't want it to be configurable.

File: core/market/src/negotiation/common.rs [hunk 32]
```diff
@@ -295,6 +379,21 @@ impl CommonBroker {
         }
         Ok(())
     }
+
+    pub async fn notify_agreement(&self, agreement: &Agreement) {
+        let session_notifier = self.session_notifier.clone();
+
+        // Notify everyone waiting on Agreement events endpoint.
+        if let Some(_) = &agreement.session_id {
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/common.rs:387]: Yes we do. But it doesn't hurt, because this is only notification and `query_agreement_events` must get events from database whatsoever. So it won't find any events for wrong identity.
But I modified comment near: https://github.com/golemfactory/yagna/pull/793#discussion_r528020727 to make it explicit

File: core/market/src/negotiation/common.rs (hunk 23 above)
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/common.rs:224]: Comment modified according to response to: https://github.com/golemfactory/yagna/pull/793#discussion_r528021680
This comment was partially true

File: core/market/src/protocol/negotiation/error.rs [hunk 33]
```diff
@@ -86,8 +86,8 @@ pub enum RemoteProposeAgreementError {
 pub enum ApproveAgreementError {
     #[error("Approve {0}.")]
     Gsb(#[from] GsbAgreementError),
-    #[error("Remote Approve: {0}")]
-    Remote(#[from] RemoteAgreementError),
+    #[error("Remote failed to approve Agreement [{1}]. Error: {0}")]
+    Remote(RemoteAgreementError, AgreementId),
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/protocol/negotiation/error.rs:90]: Hmm I would strive towards approach used in `ProposeAgreementError`, that `Remote` enum option contains `AgreementId` and `RemoteAgreementError` doesn't have to. But I'm not sure how to do this now...

File: core/market/src/db/model/agreement_events.rs (hunk 16 above)
Comment (nieznanysprawiciel) [reply core/market/src/db/model/agreement_events.rs:62]: You mean, that you would set this error message in JsonResponse?

File: core/market/tests/test_app_session_id.rs [hunk 34]
```diff
@@ -0,0 +1,549 @@
...
+        req_engine
+            .confirm_agreement(req_id.clone(), &agreement_id, Some(session_id.clone()))
+            .await
+            .unwrap();
+    }
+
+    let ids = agreements.clone();
+    let session_ids = sessions.clone();
+    let query_handle = tokio::task::spawn_local(async move {
+        for (agreement_id, session_id) in ids.iter().zip(session_ids.iter()) {
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/tests/test_app_session_id.rs:77]: It's not corner case at all. We do this on two separates nodes.

File: core/market/tests/test_app_session_id.rs [hunk 35]
```diff
@@ -0,0 +1,549 @@
...
+
+const REQ_NAME: &str = "Node-1";
+const PROV_NAME: &str = "Node-2";
+
+/// Create Agreements with different session ids. Query Agreement
+/// events using different session ids on both Provider and Requestor side.
+#[cfg_attr(not(feature = "test-suite"), ignore)]
+#[actix_rt::test]
+#[serial_test::serial]
+async fn test_session_events_filtering() -> Result<()> {
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/tests/test_app_session_id.rs:20]: See comment about corner case

File: core/market/src/config.rs [hunk 36]
```diff
@@ -39,15 +40,16 @@ impl Default for DiscoveryConfig {
 impl Default for SubscriptionConfig {
     fn default() -> Self {
         SubscriptionConfig {
-            default_ttl: chrono::Duration::hours(24),
+            default_ttl: chrono::Duration::hours(1),
```
Comment (tworec) [inline core/market/src/config.rs:43]: resolves #800 thx!

File: core/market/src/db/model/agreement.rs (hunk 14 above)
Comment (tworec) [reply core/market/src/db/model/agreement.rs:17]: you have implicitly set this limit via `up.sql` ;)

File: core/market/src/db/model/agreement_events.rs (hunk 16 above)
Comment (tworec) [reply core/market/src/db/model/agreement_events.rs:62]: yes, and extend this message to contain whole reason contents

File: core/market/src/utils/display.rs (hunk 21 above)
[the remark this answers is not in the document]
Comment (tworec) [reply core/market/src/utils/display.rs:38]: yeah, I've also realized that. But still having separate Display is cool and strange at the same time :) let it bee

File: utils/diesel-utils/src/lib.rs (hunk 22 above)
[the remark this answers is not in the document]
Comment (tworec) [reply utils/diesel-utils/src/lib.rs:6]: this information is in crate name from which this macro orignates. See 'Text` annotation from Diesel for example ;) but I am not pushing

[the remark this answers is not in the document]
Comment (tworec) [reply utils/diesel-utils/src/lib.rs:6]:
```suggestion
#[proc_macro_derive(DbTextField)]
```

File: core/market/src/negotiation/common.rs (hunk 24 above)
[the remark this answers is not in the document]
Comment (tworec) [reply core/market/src/negotiation/common.rs:430]: ok, right

File: core/market/src/negotiation/requestor.rs (hunk 25 above)
[the remark this answers is not in the document]
Comment (tworec) [reply core/market/src/negotiation/requestor.rs:490]: ok, so lets log both on debug

File: core/market/src/protocol/negotiation/error.rs (hunk 33 above)
[the remark this answers is not in the document]
Comment (tworec) [reply core/market/src/protocol/negotiation/error.rs:90]: Lets do not waste our time on it. Just remove redundancy. That's all.

File: core/market/tests/test_agreement_events.rs (hunk 26 above)
[the remark this answers is not in the document]
Comment (tworec) [reply core/market/tests/test_agreement_events.rs:75]: ok, let it be

File: core/market/tests/test_app_session_id.rs (hunk 34 above)
[the remark this answers is not in the document]
Comment (tworec) [reply core/market/tests/test_app_session_id.rs:77]: yes, exactly. And thats why it is extremely improbable that both agents will use the same sessions being independent.

File: core/market/src/negotiation/common.rs [hunk 37]
```diff
@@ -141,6 +166,71 @@ impl CommonBroker {
...
+                    NotifierError::Unsubscribed(_) => Err(AgreementEventsError::Internal(format!(
+                        "Code logic error. Shouldn't get Unsubscribe in Agreement events notifier."
+                    ))),
+                };
+            }
+            // Ok result means, that event with required sessionId id was added.
+            // We can go to next loop to get this event from db. But still we aren't sure
+            // that list won't be empty, because we could get notification for the same appSessionId,
+            // but for different identity. Of course we don't return events for other identities,
+            // so we will go to sleep again.
```
Comment (tworec) [inline core/market/src/negotiation/common.rs:230]:
```suggestion
            // so we will go to sleep ( well ;-] ) again.
```

File: core/market/src/rest_api.rs [hunk 38]
```diff
@@ -9,15 +9,18 @@ use serde::Deserialize;
...
 
 pub(crate) mod common;
 mod error;
 pub(crate) mod provider;
 pub(crate) mod requestor;
 
-const DEFAULT_EVENT_TIMEOUT: f32 = 0.0; // seconds
-const DEFAULT_QUERY_TIMEOUT: f32 = 12.0;
+const DEFAULT_EVENT_TIMEOUT: f32 = 5.0; // seconds
+const DEFAULT_QUERY_TIMEOUT: f32 = 5.0;
```
Comment (tworec) [inline core/market/src/rest_api.rs:23]: below you can find examlpe how to handle invalid user input for `appSessionId` ie. len > 100 chars in this case

File: core/market/tests/test_app_session_id.rs [hunk 39]
```diff
@@ -0,0 +1,604 @@
...
+            Some(10),
+            confirm_timestamp,
+            &req_id,
+        )
+        .await
+        .unwrap();
+
+    // Each side gets only his own event.
+    assert_eq!(p_events.len(), 2);
+    assert_eq!(r_events.len(), 2);
```
Comment (tworec) [inline core/market/tests/test_app_session_id.rs:334]: it seems a bug. Should it be `1` on both sides?

**Review by tworec — CHANGES_REQUESTED** [review]

File: core/market/tests/test_app_session_id.rs (hunk 39 above)
Comment (nieznanysprawiciel) [reply core/market/tests/test_app_session_id.rs:334]: It's what we decided. We wanted to ignore case of Provider and Requestor on the same node with the same identity, so there's no way to distinguish

File: core/market/tests/test_app_session_id.rs (hunk 34 above)
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/tests/test_app_session_id.rs:77]: Purpose of this test isn't checking what happens if session id is the same. I used same ids, because it doesn't matter and was easier to have one vectors of ids than 2

File: core/market/src/negotiation/requestor.rs (hunk 25 above)
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply core/market/src/negotiation/requestor.rs:490]: We decided to leave it on info level.

**Review by tworec — APPROVED** [review]
